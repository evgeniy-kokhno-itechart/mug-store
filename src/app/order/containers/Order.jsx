import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import OrderTable from '../components/OrderTable';
import OrderForm from './OrderForm';
import { TotalCost, submitCartOrder, cartCostsSelector } from '../../cart';

export const Order = ({
  cart,
  currentUser,
  isCurrencyLoading,
  // eslint-disable-next-line no-shadow
  submitCartOrder,
}) => (
  <React.Fragment>
    <h1 className="text-center m-3">Please check your order</h1>
    <div className="row m-2">
      <div className="col-md-6">
        <OrderTable sortColumn="title" products={cart.products} isCurrencyLoading={isCurrencyLoading} />
        <TotalCost total={cart.totalCost} customClasses="float-right" />
      </div>
      <div className="col-md-5 offset-md-1">
        <OrderForm currentUser={currentUser} onOrderSubmit={submitCartOrder} />
      </div>
    </div>
  </React.Fragment>
);

Order.propTypes = {
  cart: PropTypes.shape({
    products: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        quantity: PropTypes.number,
        price: PropTypes.object,
        title: PropTypes.string,
      }),
    ),
    totalCost: PropTypes.number,
  }).isRequired,
  currentUser: PropTypes.shape({
    id: PropTypes.string,
    address: PropTypes.string,
    city: PropTypes.string,
    country: PropTypes.string,
    name: PropTypes.string,
    phone: PropTypes.string,
    roles: PropTypes.arrayOf(PropTypes.string),
    username: PropTypes.string,
  }),

  isCurrencyLoading: PropTypes.bool,

  submitCartOrder: PropTypes.func.isRequired,
};

Order.defaultProps = {
  currentUser: {},
  isCurrencyLoading: true,
};

const mapStateToProps = state => ({
  cart: cartCostsSelector(state),
  currentUser: state.user.currentUser,
  isCurrencyLoading: state.currency.currenciesStatus.isInProcess,
});

const mapDipatchToProps = {
  submitCartOrder,
};

export default connect(
  mapStateToProps,
  mapDipatchToProps,
)(Order);
