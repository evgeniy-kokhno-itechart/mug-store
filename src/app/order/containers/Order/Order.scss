@import "Mixins";

.order {
  margin: 1rem;

  &__header {
    text-align: center;
  }

  &__content {
    display: flex;

    @include respond(phone) {
      flex-direction: column;
    }
  }

  &__summary {
    flex: 1;
    margin: 0.5rem 1.5rem 0.5rem 0.5rem;

    @include respond(phone) {
      margin: 0.5rem 1rem;
    }
  }

  &-total-cost {
    float: right;
    font-size: 1.3rem;
    font-weight: 500;
  }

  &__order-form {
    flex: 1;
    margin: 0.5rem 0.5rem 0.5rem 1.5em;

    @include respond(phone) {
      margin: 0.5rem 1rem;
    }
  }
}
