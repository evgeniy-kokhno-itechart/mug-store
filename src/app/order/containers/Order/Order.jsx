import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import OrderTable from '../../components/OrderTable/OrderTable';
import OrderForm from '../../components/OrderForm/OrderForm';
import { TotalCost, cartActions, cartCostsSelector } from '../../../cart';
import { orderConfirmedAddress } from '../../../shared';
import './Order.scss';

export class Order extends Component {
  submitOrder = () => {
    this.props.submitCartOrder(orderConfirmedAddress);
  }

  render() {
    const { cart, currentUser, isCurrencyLoading } = this.props;
    return (
      <div className="order">
        <h1 className="order__header">Please check your order</h1>
        <div className="order__content">
          <div className="order__summary">
            <OrderTable sortColumn="title" products={cart.products} isCurrencyLoading={isCurrencyLoading} />
            <TotalCost total={cart.totalCost} customClasses="order-total-cost" />
          </div>
          <div className="order-form">
            <OrderForm currentUser={currentUser} onSubmit={this.submitOrder} />
          </div>
        </div>
      </div>
    );
  }
}

Order.propTypes = {
  cart: PropTypes.shape({
    products: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        quantity: PropTypes.number,
        price: PropTypes.object,
        title: PropTypes.string,
      }),
    ),
    totalCost: PropTypes.number,
  }).isRequired,
  currentUser: PropTypes.shape({
    id: PropTypes.string,
    address: PropTypes.string,
    city: PropTypes.string,
    country: PropTypes.string,
    name: PropTypes.string,
    phone: PropTypes.string,
    roles: PropTypes.arrayOf(PropTypes.string),
    username: PropTypes.string,
  }),

  isCurrencyLoading: PropTypes.bool,

  submitCartOrder: PropTypes.func.isRequired,
};

Order.defaultProps = {
  currentUser: {},
  isCurrencyLoading: true,
};

const mapStateToProps = state => ({
  cart: cartCostsSelector(state),
  currentUser: state.user.currentUser,
  isCurrencyLoading: state.currency.currenciesStatus.isInProcess,
});

const mapDipatchToProps = {
  submitCartOrder: cartActions.SubmitCartOrder.InitiateApiCall,
};

export default connect(
  mapStateToProps,
  mapDipatchToProps,
)(Order);
