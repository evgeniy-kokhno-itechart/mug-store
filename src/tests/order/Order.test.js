/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Order } from '../../app/order/containers/Order/Order';

configure({ adapter: new Adapter() });

describe('<Order />', () => {
  let OrderWrapper;
  const fakeCart = {
    products: [{ id: '1', title: 'test product 1' }, { id: '2', title: 'test product 2' }],
    totalCost: 10,
  };

  beforeEach(() => {
    const submitCartOrder = jest.fn();
    OrderWrapper = shallow(
      <Order cart={fakeCart} submitCartOrder={submitCartOrder} />,
    );
  });

  it('renders properly with default props', () => {
    expect(OrderWrapper).toMatchSnapshot();
  });
});
