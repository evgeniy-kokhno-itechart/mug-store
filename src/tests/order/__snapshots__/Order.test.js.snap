// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Order /> renders properly with default props 1`] = `
<Fragment>
  <h1
    className="text-center m-3"
  >
    Please check your order
  </h1>
  <div
    className="row m-2"
  >
    <div
      className="col-md-6"
    >
      <Connect(OrderTable)
        products={
          Array [
            Object {
              "id": "1",
              "title": "test product 1",
            },
            Object {
              "id": "2",
              "title": "test product 2",
            },
          ]
        }
        sortColumn="title"
      />
      <TotalCostCalculator
        customClasses="float-right"
        products={
          Array [
            Object {
              "id": "1",
              "title": "test product 1",
            },
            Object {
              "id": "2",
              "title": "test product 2",
            },
          ]
        }
      />
    </div>
    <div
      className="col-md-5 offset-md-1"
    >
      <OrderForm
        currentUser={Object {}}
        onOrderSubmit={[MockFunction]}
      />
    </div>
  </div>
</Fragment>
`;
